androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/TextFields.kt

```diff
@@ -0,0 +1,46 @@
+package co.nimblehq.avishek.phong.kmmic.android.ui.screen.surveydetail
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import co.nimblehq.avishek.phong.kmmic.android.ui.common.PrimaryTextField
+import co.nimblehq.avishek.phong.kmmic.presentation.uimodel.SurveyAnswerUiModel
+
+@Composable
+fun TextFields(
+    surveyAnswerUiModels: List<SurveyAnswerUiModel>,
+    modifier: Modifier = Modifier
+) {
+    LazyColumn(
+        verticalArrangement = Arrangement.spacedBy(16.dp),
+        modifier = modifier
+    ) {
+        items(surveyAnswerUiModels.size) { index ->
+            var value by remember { mutableStateOf("") }
+
+            PrimaryTextField(
+                value = value,
+                onValueChange = {
+                    value = it
+                },
+                placeholder = surveyAnswerUiModels[index].placeholder.orEmpty(),
+                imeAction = if (index == surveyAnswerUiModels.lastIndex) ImeAction.Done else ImeAction.Next
+            )
+        }
+    }
+}
+
+@Preview
+@Composable
+fun TextFieldPreview(
+    @PreviewParameter(SurveyDetailScreenPreviewParameterProvider::class) params: SurveyDetailScreenPreviewParameterProvider.Params
```

Comment:
Even though you are getting only the answers of the first index from `PreviewParameter`, this function will run more than once, so you get 6 previews here. FYI, previews also take build times. The more preview your composable has, the more you need time to build. Please refactor a bit to reduce the number of previews here. 🙏

Reply:
Updated https://github.com/nimblehq/avishek-phong-kmm-ic/pull/124/commits/41787160bb352375fce5ee28532b76a0ae91cb9e